### ISSUE TITLE
Error: Cannot compute gradient: gradient function not found for gather. 

### ISSUE THREAD

**Issue by LunaLuan** [tensorflow/tfjs#235]
To get help from the community, check out our [Google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs).

#### TensorFlow.js version
0.10.0
#### Browser version
CLI

#### Describe the problem or feature request
I can't train tf.layer.embedding.
 **Error: (node:14028) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Cannot compute gradient: gradient function not found for
gather.
(node:14028) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate
the Node.js process with a non-zero exit code.**

#### Code to reproduce the bug / link to feature request

> import * as natural from "natural"
> import * as tf from "@tensorflow/tfjs"
> import * as processing from "./processing"
> 
> const max_length = 5
> 
> const docs = ["Well done!", "Good work", "Great effort", "nice work", "Excellent!",
>         "Weak", "Poor effort!", "not good", "poor work", "Could have done better."]
> const labels = [1, 1, 1, 1, 1, 0, 0, 0, 0, 0]
> 
> 
> 
> export function transfor(arr) {
>     let x = []
>     
>     arr.forEach((a) => {
>         const t = tf.tensor1d(a)
>         x.push(t.pad([[0, max_length - a.length]]))
>     })
> 
>     return tf.stack(x)
> }
> 
> const arr = processing.encode(docs)
> 
> const x = transfor(arr)
> const y = tf.tensor(processing.getCategories(labels, 2))
> 
> 
> const model = tf.sequential()
> // model.add(tf.layers.dense({
> //     inputShape: [5],
> //     units: 1
> // }))
> model.add(tf.layers.embedding({
>     inputDim: processing.getDictionarySize(),
>     outputDim: 4,
>     inputLength: max_length
> }))
> model.add(tf.layers.flatten())
> model.add(tf.layers.dense({
>     units: 2,
>     activation: "softmax"
> }))
> 
> model.compile({
>     optimizer: tf.train.adam(0.01),
>     loss: "categoricalCrossentropy",
>     metrics: ["accuracy"]
> })
> 
> model.predict(tf.tensor([[1, 2, 0, 0, 0]])).print()
> 
> async function train() {
>     for(let i = 0; i < 100; i++) {
>         const history = await model.fit(
>             x,  y, 
>             {
>                 batchSize: 4,
>                 epochs: 4
>             }
>         )
>         console.log("Loss after Epoch "+ i  + ": "+ history.history.loss[0])
>         console.log("\tAccurancy: " + history.history.acc[0])
>     }
> }
> 
> train()
> 
>

**Comment by ManrajGrover** [comment]
@LunaLuan We are tracking gradient for gather in #135 which will get fixed in https://github.com/tensorflow/tfjs-core/pull/981

**Comment by tafsiri** [comment]
closed by https://github.com/tensorflow/tfjs-core/pull/981
